e2e/pageobjects/ide/Editor.ts

```diff
@@ -65,74 +93,118 @@ export class Editor {
         await this.driverHelper.waitAndClick(tabCloseButtonLocator, timeout);
     }
 
-    async waitEditorOpened(timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+    async waitEditorOpened(editorTabTitle: string, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
         const firstEditorLineLocator: By = By.xpath(this.getEditorLineXpathLocator(1));
 
-        await this.driverHelper.waitVisibility(By.css(Editor.EDITOR_BODY_CSS), timeout);
-        await this.driverHelper.waitVisibility(firstEditorLineLocator, timeout);
+        await this.driverHelper.waitPresence(this.getEditorBodyLocator(editorTabTitle), timeout);
+        await this.driverHelper.waitPresence(firstEditorLineLocator, timeout);
     }
 
     async waitEditorAvailable(tabTitle: string, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
         await this.waitTab(tabTitle, timeout);
-        await this.waitEditorOpened(timeout);
+        await this.waitEditorOpened(tabTitle, timeout);
     }
 
     async getLineText(lineNumber: number): Promise<string> {
         const lineIndex: number = lineNumber - 1;
-        const editorText: string = await this.getEditorText();
+        const editorText: string = await this.getEditorVisibleText();
         const editorLines: string[] = editorText.split('\n');
         const editorLine = editorLines[lineIndex] + '\n';
 
         return editorLine;
     }
 
-    async getEditorText(): Promise<string> {
-        const interactionContainerLocator: By = By.css(Editor.EDITOR_INTERACTION_СSS);
-
-        const editorText: string = await this.driverHelper.waitAndGetValue(interactionContainerLocator);
+    async getEditorVisibleText(): Promise<string> {
+        const editorBodyLocator: By = By.xpath('//div[@class=\'view-lines\']');
+        const editorText: string = await this.driverHelper.waitAndGetText(editorBodyLocator);
         return editorText;
     }
 
-    async moveCursorToLineAndChar(line: number, char: number) {
+    async waitText(expectedText: string,
+        timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT,
+        polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING) {
+        await this.driverHelper.getDriver().wait(async () => {
+            const editorText: string = await this.getEditorVisibleText();
+            const isEditorContainText: boolean = editorText.includes(expectedText);
+
+            if (isEditorContainText) {
+                return true;
+            }
+
+            this.driverHelper.wait(polling);
+        }, timeout);
+    }
+
+    async waitForText(editorTabTitle: string,
```

Comment:
Hmm... now, the method is called "generally", but it's still waiting for some specific, hard-coded text.
My original suggestion was to rewrite the method to take string parameter (text to wait for).

Now I can also see we already have such method (`waitText`), but that one has slightly different logic (it's not scrolling down, as this one).
Maybe this one shoud be renamed&rewritten to something like 
```
followAndWaitForText(editorTabTitle: string, textToSearchFor: string, timeout: number, polling:number)
```